### PR TITLE
Introduce actionlint job for linting GitHub Actions workflows

### PR DESCRIPTION
**Pull request by tfrommen**
This is a suggestion to address #84, using the reusable `lint-workflows` workflow, which under the hood is based on [actionlint](https://github.com/rhysd/actionlint).

I kept the workflow trigger the same as for code quality.

### CODE REVIEW

**Comment by swissspidy** [comment]
Thanks for the PR, @tfrommen!

I don't think we'd want to use `inpsyde/reusable-workflows/.github/workflows/lint-workflows.yml@main` here, but either add those two steps directly to the repo or use the [suggested steps from the projects' example](https://github.com/rhysd/actionlint/blob/main/docs/usage.md#use-actionlint-on-github-actions) (with [problem matchers](https://github.com/rhysd/actionlint/blob/main/docs/usage.md#problem-matchers)).

Also, if we move this to `reusable-code-quality.yml`, then we can automatically use this workflow here and in all other repos.

**Comment by tfrommen** [comment]
Hi @swissspidy, happy to change things. What should it be?

**Comment by swissspidy** [comment]
After sleeping on it, let's go with the suggested integration as per the project's examples, and do that in `reusable-code-quality.yml`

**Comment by tfrommen** [comment]
@swissspidy done. I added the `shellcheck` integration, consideirng that there are some bash/shell scripts included in this repository already, and I assume more in others.

**Review by swissspidy — APPROVED** [review]
Lets' try this out

**Comment by jrfnl** [comment]
> I added the `shellcheck` integration, considering that there are some bash/shell scripts included in this repository already, and I assume more in others.

I get the impression that shellcheck is now _disabled_ instead... ?

From `actionlint -h`:
```bash
Flags:
  -shellcheck string
        Command name or file path of "shellcheck" external command. If empty, shellcheck integration will be disabled (default "shellcheck")
```

File: .github/workflows/reusable-code-quality.yml [hunk 1]
```diff
@@ -11,6 +11,23 @@ concurrency:
...
 
+  actionlint: #-----------------------------------------------------------------------
+    name: Lint GitHub Actions workflows
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v4
+
+      - name: Add problem matcher
+        run: |
```
Comment (jrfnl) [inline .github/workflows/reusable-code-quality.yml:22]: Suggestion:
```yaml
        if: ${{ github.event_name == 'pull_request' }}
```

**Pull request merged**